components/places/src/ffi.rs

```diff
@@ -43,6 +48,155 @@ impl UniffiCustomTypeWrapper for Url {
     }
 }
 
+impl UniffiCustomTypeWrapper for Guid {
+    type Wrapped = String;
+
+    fn wrap(val: Self::Wrapped) -> uniffi::Result<Guid> {
+        Ok(Guid::new(val.as_str()))
+    }
+
+    fn unwrap(obj: Self) -> Self::Wrapped {
+        obj.into()
+    }
+}
+
+impl UniffiCustomTypeWrapper for Timestamp {
+    type Wrapped = i64;
+
+    fn wrap(val: Self::Wrapped) -> uniffi::Result<Timestamp> {
+        Ok(Timestamp::from(val as u64))
+    }
+
+    fn unwrap(obj: Self) -> Self::Wrapped {
+        obj.as_millis() as i64
+    }
+}
+
+///*
+/// A bookmark node.
+///
+/// We use a single message type for bookmarks. It covers insertion, deletion,
+/// and update, and represents all three bookmark types.
+///
+/// This simplifies the FFI by reducing the number of types that must go across
```

Comment:
I think it's worth updating this comment to note something like "when we had hand-written FFIs"